framework/jquery/ActiveFormClientScript.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\jquery;
+
+use Yii;
+use yii\helpers\Json;
+
+/**
+ * ActiveFormClientScript is a behavior for [[\yii\widgets\ActiveForm]], which allows composition
+ * of the client-side and AJAX form validation via underlying JQuery plugin.
+ *
+ * Usage example:
+ *
+ * ```php
+ * <?php $form = \yii\widgets\ActiveForm::begin([
+ *     'id' => 'example-form',
+ *     'as clientScript' => \yii\jquery\ActiveFormClientScript::class,
+ *     // ...
+ * ]); ?>
+ * ...
+ * <?php \yii\widgets\ActiveForm::end(); ?>
+ * ```
+ *
+ * @see \yii\widgets\ActiveForm
+ * @see \yii\widgets\ActiveFormClientScriptBehavior
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @since 2.1.0
+ */
+class ActiveFormClientScript extends \yii\widgets\ActiveFormClientScript
+{
+    /**
+     * @inheritdoc
+     */
+    public $clientValidatorMap = [
```

Comment:
I suggest to move this array to a separate method `getClientValidatorMap()` to make it easier to add custom validators to this map, or override existing.

Reply:
Or, maybe, create a `ClientValidatorFactory` and inject it in `ActiveFormClientScript`.
It will be responsible for creating `ClientValidator` based on `Validator`, than it should be easier to add new mappings

Reply:
`defaultClientValidatorMap()` added